### ISSUE TITLE
[bug] icon size, task width both calculated incorrectly if different scaling factors are used on different displays 

### ISSUE THREAD

**Issue by FFdhorkin** [Drugwash2/multi-line-taskbar#4]
There appears to be an issue with how icon size is calculated.

It appears to be an issue with this line of code:
`this.iconSize = Math.floor((panel_height - this._applet._preferences.taskbar_row_count + 1) / this._applet._preferences.taskbar_row_count * ICON_HEIGHT_FACTOR * global.ui_scale);`

It appears that `global.ui_scale` does not give the current display's scaling factor so much as an arbitrary scaling factor.

My primary (vertical) monitor has monitor scale 100%, but my secondary (4K) monitor has monitor scale 125%. As a result, various UI calculations are wrong. (Incidentally, this appears to be the cause of the issue with alignment)

As a workaround, I simply removed the `global.ui_scale` and now the `Task Icon Size` setting is being respected properly on my 100% scaling monitor.

With 8 buttons per row:
![image](https://user-images.githubusercontent.com/59899746/199817386-77cd69ef-7c69-4f4a-85bd-da4a81a063a7.png)

With 6 buttons per row:
![image](https://user-images.githubusercontent.com/59899746/199817603-2d1e0d6c-3aae-4dde-b5c7-d34db1c4720f.png)

With 4 buttons per row:
![image](https://user-images.githubusercontent.com/59899746/199817699-6a18067a-6389-495c-a49a-1745a6c7bdde.png)

With 3 buttons per row:
![image](https://user-images.githubusercontent.com/59899746/199817841-f5c2e892-7bf8-4201-986f-fa06a370c3ae.png)

**Comment by FFdhorkin** [comment]
As a side note, it would be nice to dynamically set icon size / row height based on how many windows you have open. i.e. the following pseudo:
```
const numRows = Math.min(numWindows / buttonsPerRow, maxRows);
const rowHeight = panelHeight / numRows;
```

**Comment by Drugwash2** [comment]
> There appears to be an issue with how icon size is calculated.
Yeah, that's been bugging me since forever. I tried to mess with the calculations in the past but there were more variables at play and never managed to get the optimal result.

`global.ui_scale` should simply be a value of **1** for a 100% scaling display. However the mere usage of the word `global` hints at not using individual display scaling. I'm not yet sure how to correctly retrieve the real scaling for each monitor, and unfortunately I also don't have a multi-monitor setup at hand for testing - this notebook acts very badly at adding/removing an external monitor under Mint 19.2, it requires a full reboot to get it back in shape so I avoid trying that. The test Mint 21 machine has a single VGA video output so no luck there either.

Your idea is good in principle but it would invalidate the fixed rows number setting. So again there should be an additional switch to select between fixed number of rows or dynamic adjustment in order to fit any user preference. Will take it into consideration.

**Comment by Drugwash2** [comment]
Finally found out a way to allocate only the necessary width to any button even when there's plenty of space left, so there won't be any very wide buttons anymore regardless of the max allowed width (`allocNS` value).
Also implemented your dynamic row height.
Icon size issue is still unsolved but will keep working on it.
In the mean time please test this quick alpha version: 
[multilineWL@drugwash 1.2.4a1.zip](https://github.com/Drugwash2/multi-line-taskbar/files/9938144/multilineWL%40drugwash.1.2.4a1.zip)
